dart/lib/src/protocol/sentry_span.dart

```diff
@@ -44,14 +46,23 @@ class SentrySpan extends ISentrySpan {
     if (status != null) {
       _status = status;
     }
-    _timestamp = endTimestamp ?? getUtcDateTime();
+
+    if (endTimestamp?.isBefore(_startTimestamp) ?? false) {
+      _hub.options.logger(
+        SentryLevel.warning,
+        'End timestamp ($endTimestamp) cannot be before start timestamp ($_startTimestamp)',
+      );
+      _endTimestamp = getUtcDateTime();
+    } else {
+      _endTimestamp = endTimestamp?.toUtc() ?? getUtcDateTime();
```

Comment:
`final newTimeStamp = endTimestamp?.toUtc() ?? getUtcDateTime();`

This could be done before and here you just assign the value to the `_endTimestamp` var so we have the `endTimestamp` var computed as early as possible in the finish method call.

Reply:
You'd not assign to `_endTimestamp`, rather to a new var e.g. `newTimeStamp` as my code snippet and only later assign to `_endTimestamp` depending on the `if` condition result.
The idea is just to compute `getUtcDateTime()` as early as possible.
Just an early `final utcDateTime = getUtcDateTime()` is already good enough instead of calling multiple times `getUtcDateTime()` later.